### PR TITLE
Fix a couple of bugs

### PR DESCRIPTION
**Pull request by posobin**

### CODE REVIEW

File: plugins/deepgram/src/stt.ts [hunk 1]
```diff
@@ -261,12 +261,12 @@ export class SpeechStream extends stt.SpeechStream {
                   if (isFinal) {
                     this.queue.put({
                       type: stt.SpeechEventType.FINAL_TRANSCRIPT,
-                      alternatives: [alternatives[0], ...alternatives.splice(0)],
```
Comment (posobin) [inline plugins/deepgram/src/stt.ts:264]: This just duplicates the first item in the array. Not sure what the intention was, perhaps to shallowly copy the array?

File: agents/src/pipeline/pipeline_agent.ts [hunk 2]
```diff
@@ -693,31 +693,31 @@ export class VoicePipelineAgent extends (EventEmitter as new () => TypedEmitter<
...
 
-      if (handle.addToChatCtx && (!userQuestion || handle.userCommitted)) {
-        this.chatCtx.messages.push(...extraToolsMessages);
-        if (interrupted) {
-          collectedText + '…';
-        }
-
-        const msg = ChatMessage.create({ text: collectedText, role: ChatRole.ASSISTANT });
-        this.chatCtx.messages.push(msg);
+    if (handle.addToChatCtx && (!userQuestion || handle.userCommitted)) {
```
Comment (posobin) [inline agents/src/pipeline/pipeline_agent.ts:698]: This was nested in an if when I think it wasn't supposed to be so AGENT_SPEECH_INTERRUPTED/COMMITTED were never firing

**Review by theomonnom — APPROVED** [review]
Thanks!

File: plugins/deepgram/src/stt.ts [hunk 3]
```diff
@@ -261,12 +261,14 @@ export class SpeechStream extends stt.SpeechStream {
...
+                      // Ensures that TS knows that the array is not empty
```
Comment (nbsp) [inline plugins/deepgram/src/stt.ts]: can you delete these comments? these are helpful at review time but not so much as code

**Review by nbsp — APPROVED** [review]
looks good, thanks! just one nitpick and i'll merge it.

File: plugins/deepgram/src/stt.ts (hunk 3 above)
Comment (posobin) [reply plugins/deepgram/src/stt.ts]: I don't agree (would have been useful to me when reading this code), but sure, removed

**Pull request merged**